source/fe/fe_values.cc

```diff
@@ -2674,7 +2676,8 @@ namespace internal
     AssertDimension(values.size(), n_quadrature_points);
 
     // initialize with zero
-    std::fill_n (values.begin(), n_quadrature_points, dealii::internal::NumberType<typename std::decay<Number>::type>::value(0.0));
+    std::fill_n (values.begin(), n_quadrature_points,
+                 dealii::internal::NumberType<typename std::decay<Number>::type>::value(0.0));
```

Comment:
Unrelated to the current patch, but do you actually need the `std::decay` here and in line 2943 below?

Reply:
It's not terribly important, but if you had a minute, it would be nice to see the error message to understand the issue. I suspect it's the same as with `ProductType`, namely that if `Number` happens to be CV qualified, then the provided specializations no longer work. We know now how one could work around this, though it's less clear to me that that's worth it given that everything is already only used internally.